src/main/java/sx/blah/discord/handle/impl/obj/Guild.java

```diff
@@ -779,6 +778,18 @@ public IEmoji getEmojiByName(String name) {
 	}
 
 	@Override
+	public IEmoji addEmoji(String name, BufferedImage image){
+		if (name.length() < 2 || name.length() > 32 || DiscordUtils.EMOJI_CREATE_NAME.matcher(name).find())
+			throw new DiscordException("Emoji names must be at least 2 characters long and can only contain alphanumeric characters and underscores.");
+
+		DiscordUtils.checkPermissions(client, this, EnumSet.of(Permissions.MANAGE_EMOJIS));
+
+		DataBufferByte data = (DataBufferByte) image.getRaster().getDataBuffer();
+		EmojiObject response = ((DiscordClientImpl) client).REQUESTS.POST.makeRequest(DiscordEndpoints.GUILDS + getID() + "/emojis", new EmojiCreateRequest(name, data.getData()), EmojiObject.class);
```

Comment:
Because the Permissions.MANAGE_EMOJIS and whitelist are separate, catching the exception thrown by DiscordUtils.checkPermissions wouldn't be very helpful, and I was told that the response would just silently error. The response returned by Discord is as follows:

```json
{
  "code": 20005,
  "message": "Bots normally cannot create emojis. If you are building a service that needs to create emojis please contact support@discordapp.com to be whitelisted."
}
```

Would this response throw a DiscordException?